shared/challenge_builder.py

```diff
@@ -0,0 +1,299 @@
+        if self.type != "Polygon":
+            raise ValueError("This function only works for Polygons")
+        self.type = "LineString"
+        self.coordinates = self.coordinates[0]
+
+    def convertClosedStringToPolygon(self):
+        if self.type != "LineString":
+            raise ValueError("This function only works for LineStrings")
+        self.type = "Polygon"
+        self.coordinates = [self.coordinates]
+
+    def getCenterPoint(self):
+        if self.type == "Point":
+            return self.coordinates
+        elif self.type == "LineString":
+            # Create a center point as an average of all points
+            lat = 0
+            lon = 0
+            for point in self.coordinates:
+                lat += point[1]
+                lon += point[0]
+            return [lon / len(self.coordinates), lat / len(self.coordinates)]
+        elif self.type == "Polygon":
+            # Create a center point as an average of all points
+            lat = 0
+            lon = 0
+            for point in self.coordinates[0]:
+                lat += point[1]
+                lon += point[0]
+            return [lon / len(self.coordinates[0]), lat / len(self.coordinates[0])]
+
+@dataclass
+class GeoFeature:
+    # This is an abstraction of a GeoJSON Feature
+    # A geo feature is a feature that has a geometry and properties
+    # The geometry is a dict with a type and coordinates
+    # The properties is a dict with key-value pairs
+    def __init__(self, geometry, properties = {}):
+        self.geometry = geometry
+        self.properties = properties
+        # infer the type of the geometry (Point, LineString, Polygon) from the coordinates
+        # if its just a list of 2 numbers, its a Point
+        # if its a list of lists of 2 numbers, its a LineString
+        # if its a list of lists of 2 numbers where start and end are the same, its a Polygon
+        if not isinstance(geometry, Geometry):
+            raise ValueError("geometry must be an instance of the Geometry class, got " + str(type(geometry)) + " instead")
+
+
+    @classmethod
+    def withId(cls, osmType, osmId, geometry, properties):
+        print(osmType, osmId, geometry, properties)
+        properties["@id"] = str(osmType) + "/" + str(osmId)
+        return cls(geometry, properties)
+
+    def toGeoJSON(self):
+        return {
+            "type": "Feature",
+            "geometry": self.geometry.toGeoJSON(),
+            "properties": self.properties
+        }
+
+    def convertPolygonToClosedString(self):
+        if self.geometryType != "Polygon":
+            raise ValueError("This function only works for Polygons")
+        self.geometryType = "LineString"
+        self.geometry = self.geometry[0]
+
+    def convertClosedStringToPolygon(self):
+        if self.geometryType != "LineString":
+            raise ValueError("This function only works for LineStrings")
+        self.geometryType = "Polygon"
+        self.geometry = [self.geometry]
```

Comment:
Please remove these functions.
Geometry manipulations should directly happen on the geometry object.